apps/src/applab/applab.js

```diff
@@ -337,6 +344,26 @@ function onDebugInputKeyDown(e) {
       outputApplabConsole('< (not running)');
     }
   }
+  if (e.keyCode == KeyCodes.UP) {
+    if (Applab.dc.i > 0) {
+      Applab.dc.i -= 1;
+    }
+    if (typeof Applab.dc.hist[Applab.dc.i] !== 'undefined') {
```

Comment:
I can change this is you would like, but this is to catch the edge cases where `i = 0` for up and `i = hist.length` for down; keeps it consistent between the two.